src/lib/services/access-service.ts

```diff
@@ -359,20 +359,28 @@ export class AccessService {
     async getProjectUsersForRole(
         roleId: number,
         projectId?: string,
-    ): Promise<IUser[]> {
-        const userIdList = await this.store.getProjectUserIdsForRole(
+    ): Promise<IProjectUser[]> {
+        const userRoleList = await this.store.getProjectUsersForRole(
             roleId,
             projectId,
         );
-        if (userIdList.length > 0) {
-            return this.userStore.getAllWithId(userIdList);
+        if (userRoleList.length > 0) {
+            const userIdList = userRoleList.map((u) => u.userId);
+            const users = await this.userStore.getAllWithId(userIdList);
+            return users.map((user) => {
+                const role = userRoleList.find((r) => r.userId == user.id);
+                return {
+                    ...user,
+                    addedAt: role.addedAt,
+                };
+            });
```

Comment:
I know this is how we've done it, but this looks to me like something that can be moved to the database layer.
Worth it to have a look later, that way you don't do two separate round-trips to the database.